### PR TITLE
Improve `mpf.__format__`

### PR DESCRIPTION
**Pull request by javierelpianista**
* Adds support for different kinds of rounding
* Adds documentation

### CODE REVIEW

File: mpmath/ctx_mp_python.py [hunk 1]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
         return t ** s
 
     def __format__(s, format_spec):
+        """
+        ``mpf`` objects allow for formatting similar to Python floats:
+
+            >>> from mpmath import pi, mp
+            >>> mp.dps = 100
+            >>> print('{:105.100f}'.format(pi()))
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:364]: ``()`` is redundant

File: mpmath/ctx_mp_python.py [hunk 2]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+               3.1415926535897932384626433832795028841971693993751058209749445923078164062862089986280348253421170680
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:365]: Probably you could cut this line per PEP 8.

File: mpmath/ctx_mp_python.py [hunk 3]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+            >>> print(f"{10*pi():.20e}")
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:366]: redundant parens too

File: mpmath/ctx_mp_python.py [hunk 4]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+            3.14159265358979323846e+01
+
+        The format specification adopts the same general form as `Python's
+        Format Specification Mini-Language <FS_>`_. The only format types
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:370]: shouldn't this be ``<_FS>``?

File: mpmath/ctx_mp_python.py [hunk 5]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+        supported by ``mpf`` are ``"e"``, ``"E"``, ``"f"``, ``"F"``, ``"g"`` ,
+        and ``"G"``.
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:372]: I would rather announce that *all* format types, valid for floating-point numbers in Python are supported.

File: mpmath/ctx_mp_python.py [hunk 6]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+
+        The formatter does not track the actual precision of the ``mpf``
+        objects. If precision is left as default, the resulting string is
+        exactly the same as if printing a regular ``float``::
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:376]: You can use intersphinx to reference actual float type docs.

File: mpmath/ctx_mp_python.py [hunk 7]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+
+            >>> mp.dps = 15
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:378]:
```suggestion
            >>> mp.dps = fp.dps
```

File: mpmath/ctx_mp_python.py [hunk 8]
```diff
@@ -356,6 +356,64 @@ def __rpow__(s, t):
...
+
+            * 'U': rounding towards plus infinity
+            * 'D': rounding towards minus infinity
+            * 'Z': rounding towards zero
+            * 'N': rounding to nearest (default)
+
+        The rounding option must be set before the presentation type. For
+        example::
+
+            >>> print('{:.5Uf}'.format(mp.mpf('1.2345678')))
```
Comment (skirpichev) [inline mpmath/ctx_mp_python.py:398]: I don't think we need so many examples here.  One-two is enough.

**Review by skirpichev — COMMENTED** [review]
This is marked a draft, so - just a couple of comments.

Maybe you should split this pr to rounding support and docs.

**Comment by javierelpianista** [comment]
@skirpichev I've split this PR into two as you suggested. I'll make a PR with the upgraded documentation when we close/merge my rounding_types one.

**Pull request closed without merging**